extract_game_files.sh

```diff
@@ -1,13 +1,38 @@
 #!/bin/bash
 
 if [ ! -f DepotDownloader ]; then
-    wget https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.0.0/DepotDownloader-linux-x64.zip -O DepotDownloader-linux-x64.zip
-    unzip -o DepotDownloader-linux-x64.zip DepotDownloader && rm DepotDownloader-linux-x64.zip
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        curl -L -o DepotDownloader.zip "https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.2.0/DepotDownloader-macos-arm64.zip"
+    elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        wget https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.0.0/DepotDownloader-linux-x64.zip -O DepotDownloader.zip
+    else
+        echo "Unsupported OS: $OSTYPE"
+        exit 1
+    fi
```

Comment:
```suggestion
    if command -v curl >/dev/null 2>&1; then
        curl -L -o DepotDownloader.zip "https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.2.0/DepotDownloader-macos-arm64.zip"
    elif command -v wget >/dev/null 2>&1; then
        wget https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.0.0/DepotDownloader-linux-x64.zip -O DepotDownloader.zip
    else
        echo "Neither wget nor curl were found."
        exit 1
    fi
```
What if we just check whichever command is available?

Reply:
DepotDownloader and Source2-CLI both have separate releases for macOS, The primary concern was mainly downloading the correct release rather than using wget vs. curl.

Linux release: https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.0.0/DepotDownloader-linux-x64.zip

MacOS release: https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.2.0/DepotDownloader-macos-arm64.zip

Reply:
Ah right I missed the different urls, just saw wget/curl.